README.md

```diff
@@ -1 +1,119 @@
+
+# Usage
+
+The goal of Gandalf was to add this basic boiler plate code you any application quickly. You will need to add the following code to your application as well as host a json file on a publicly accessible server with the format included below.
+
+#### Application Class
+
+Extend the Android [`Application`](http://developer.android.com/reference/android/app/Application.html) class and add the following to the `onCreate()`
+
+```java
+@Override
+public void onCreate() {
+    super.onCreate();
+    new Gandalf.Installer()
+            .setContext(this)
+            .setPackageName("com.my.package")
+            .setBootstrapUrl("http://www.example.com/bootstrap.json)
+            .install();
+}
+```
+
+#### Splash Activity
+
+Extend `GandalfActivity` for use as your main "Splash" type activity, this is where the magic will happen. Just provide a layout resource id to display while the bootstrap file is being checked and implement the `youShallPass()` method with what should happen after a successful check.
+
+```java
+public class SplashActivity extends GandalfActivity {
+
+    @Override
+    public void youShallPass() {
+        //After a successful bootstrap check we change the content view, you may also load a new activity or do whatever logic you want after the check is complete.
+        setContentView(R.layout.activity_splash_finished_loading);
+    }
+
+    @Override
+    public int contentView() {
+        //While the bootstrap check is running we provide a layout to be displayed
+        return R.layout.activity_splash_loading;
+    }
+}
+```
+
+#### Manifest Changes
+
+Add the `android:name` attribute to the `application` tag and specify the path to your custom [`Application`](http://developer.android.com/reference/android/app/Application.html) class from above and set your `SplashActivity` as the entry point for your app. 
+
+```xml
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="io.github.btkelly.gandalf.example">
+
+    <application
+        android:name=".CustomApplication"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:theme="@style/AppTheme">
+
+        <activity
+            android:name=".SplashActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+
+    </application>
+
+</manifest>
+```
+
+#### Json File
+
+You must host a json file remotely and set the url of this file in the the Gandalf installer. The Json file must match the following format and use the Android `versionCode` not the `versionName` for version information.
+
+```json
+{
+  "android": {
+    "alert": {
+      "message": "We are currently performing server maintenance. Please try again later.",
+      "blocking": true
+    },
+    "optionalUpdate": {
+      "optionalVersion": "6",
```

Comment:
Hmm, maybe better to use `optionalVersionCode` here to reinforce the requirement stated above?